Spigot-Server-Patches/0730-Add-Mob-lookAt-API.patch

```diff
@@ -0,0 +1,133 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <blake.galbreath@gmail.com>
+Date: Fri, 14 May 2021 13:42:17 -0500
+Subject: [PATCH] Add Mob#lookAt API
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/EntityInsentient.java b/src/main/java/net/minecraft/world/entity/EntityInsentient.java
+index a246edd09854dabf095da75c9d200f5cf26e7138..1b8f42780cf852b287a7db401e63bdb9b3d097b7 100644
+--- a/src/main/java/net/minecraft/world/entity/EntityInsentient.java
++++ b/src/main/java/net/minecraft/world/entity/EntityInsentient.java
+@@ -852,14 +852,17 @@ public abstract class EntityInsentient extends EntityLiving {
+ 
+     protected void mobTick() {}
+ 
++    public int getLookPitchSpeed() { return O(); } // Paper - OBFHELPER
+     public int O() {
+         return 40;
+     }
+ 
++    public int getBodyYawSpeed() { return Q(); } // Paper - OBFHELPER
+     public int Q() {
+         return 75;
+     }
+ 
++    public int getLookYawSpeed() { return ep(); } // Paper - OBFHELPER
```

Comment:
The first two here are *maximum* rotation values, so how many degrees they can move at the highest (but not necessarily by that amount)

Reply:
Perhaps more in the sense of `getMaxLegalPitch`? It should still be possible to set it higher with plugins and whatnot, but it ain't legal

Reply:
Its a max speed because the look controller uses these values as a clamp per tick.

Reply:
Yeah I'm not talking about the last, that's correct, I just placed the review point badly.
But: "The **first two** here are maximum rotation values [...]"

Reply:
They's _all_ speeds. Each one gets applied per tick with a max clamp of one of these values. It's applied in these increments over each tick until the desired value is finally reached. These are not limits themselves, other than to clamp each tick for speed/animation control.

Reply:
Here's the obfuscated code that can explain it better, maybe.

```java
this.a.aC = this.a(this.a.aC, this.h(), this.b);
this.a.pitch = this.a(this.a.pitch, this.g(), this.c);
```

fields:
`a.aC` is yaw
`h()` is the final yaw desired
`g()` is the final pitch desired
`b` is the yaw speed
`c` is the pitch speed

method:
`a(current, desired, speed)` this calulates the end result of _this tick_ moving the current value towards the desired value at the max speed per tick.

Reply:
I see, thanks. I guess what I'm still caught up on is some word of change missing, like delta/rotation or similar instead of "lookSpeed", or as the docs already say "yaw/pitch rotation speed". So I would suggest just adding "rotation" before "speed" in API, so it's clearer this is about head rotation, if that's fine with you.

Reply:
Sure, I'll change it to `getHeadYawSpeed` even though I feel that makes it _more_ ambiguous seeing these are _only_ used for looking. Might give off the impression this is used for other head yaw changes elsewhere in the server/api. But you're the boss. As for putting the word rotation, that's just plain redundant and yaw and pitch _are_ rotations.

Reply:
That works. I don't think it makes it more ambiguous, and adding rotation wouldn't be redundant either. "Rotation" being both a verb and noun might make it look like it were, but using similar phrases as an example, leaving it out is like saying "angle speed" or "position speed", describing the current state, not the movement - there's no verb or change (in contrast to "angle change speed", "yaw change speed", or "yaw rotation speed", rotation being the *verb/action* here).
But I'll merge after anyways, it's not that big of a deal, just a picky semantics game

Reply:
I got you fam